search/Knuth-Morris-Pratt (KMP) Algorithm for Pattern Searching

```diff
@@ -0,0 +1,57 @@
+#include <bits/stdc++.h>
```

Comment:
thank you for contributing.
there are several issues which needs to be addressed before we start reviewing.


1. file doesnt have a .cpp extension.
2. you have included bits/stdc++ which is not portable
3. using namespace std is not allowed on this repository because it is bad practice